src/vellum/utils/templating/render.py

```diff
@@ -14,6 +14,27 @@ def finalize(obj: Any) -> str:
     return str(obj)
 
 
+def custom_replace(s, old, new):
+    """
+    Custom replace filter that uses DefaultStateEncoder for conversion
+    """
+    import json
+
+    from vellum.workflows.state.encoder import DefaultStateEncoder
+
+    def encode(obj):
+        try:
+            return json.dumps(obj, cls=DefaultStateEncoder)
```

Comment:
[blocking] I don't think we _always_ want to do the json dumps - a string will go from `"hello"` to `'"hello"'`. We likely want to have the same `if not isinstance()` check